### PR TITLE
feat (DOCS-CODE-013): [api] canonical protected api access another protected resources on behalf user

### PR DESCRIPTION
**Pull request by ferantivero**
closes: #508177

### CODE REVIEW

File: src/protected-api-access-protected-api/README.md
Comment (ckittel) [inline src/protected-api-access-protected-api/README.md:25]: Where in the code is this line coming from?

File: src/protected-api-access-protected-api/README.md
Comment (ckittel) [inline src/protected-api-access-protected-api/README.md:81]: Why does this need a platform type?  An API typically wouldn't as it's not redirecting users anywhere.

File: src/protected-api-access-protected-api/README.md
Comment (ckittel) [inline src/protected-api-access-protected-api/README.md:100]: Why would the client need to create this?  It would be the API's permission that it adds.

File: src/protected-api-access-protected-api/README.md
Comment (ckittel) [inline src/protected-api-access-protected-api/README.md:120]: Does it matter if the right side of these looks different than the right side of them in the code?

File: src/protected-api-access-protected-api/Program.cs
Comment (ckittel) [inline src/protected-api-access-protected-api/Program.cs:13]: Any chance we can name this more meaningful?  Like `GraphApi` instead?

**Review by ckittel — CHANGES_REQUESTED** [review]
Couple small items, but nothing game changing. Nice work.

File: src/protected-api-access-protected-api/README.md
Comment (ferantivero) [reply src/protected-api-access-protected-api/README.md:120]: add a comma at the end as well in the `TenantId` and capitalize the `Here` word.

File: src/protected-api-access-protected-api/Program.cs
Comment (ferantivero) [reply src/protected-api-access-protected-api/Program.cs]: good call @ckittel 

done | addressed [b40f3](https://github.com/Azure-Samples/ms-identity-docs-code-dotnet/pull/13/commits/b40f3d67a1595d8fc5059a859f5f5d7ef5b03842)

File: src/protected-api-access-protected-api/README.md
Comment (ferantivero) [reply src/protected-api-access-protected-api/README.md]: good catch @ckittel,

done | addressed from [773d3](https://github.com/Azure-Samples/ms-identity-docs-code-dotnet/pull/13/commits/773d30d4cdd1a1aa0cbf2446a4bf4ae44b46d241)

File: src/protected-api-access-protected-api/README.md
Comment (ferantivero) [reply src/protected-api-access-protected-api/README.md]: thanks for the feedback @ckittel, copy pasta

done  | addressed from [9f9b8](https://github.com/Azure-Samples/ms-identity-docs-code-dotnet/pull/13/commits/9f9b8bc6efe81bff70f00c1676c5b6036cdbe1b3)

File: src/protected-api-access-protected-api/README.md
Comment (ckittel) [reply src/protected-api-access-protected-api/README.md]: They are aligned now. Looks good.

File: src/protected-api-access-protected-api/README.md
Comment (ferantivero) [reply src/protected-api-access-protected-api/README.md]: hopefully we did better now @ckittel, but open to keep iterating till we find the version we like the most


done | addressed from [caf71](https://github.com/Azure-Samples/ms-identity-docs-code-dotnet/pull/13/commits/caf712624342176207e6b30b7a670b232576b441)

Comment (ckittel) [reply src/protected-api-access-protected-api/README.md]: That helps add clarity, IMO. Thanks.

File: src/protected-api-access-protected-api/README.md
Comment (ckittel) [reply src/protected-api-access-protected-api/README.md]: No more platform type.

Also, the "Expose an API" section seems to look align with other approved examples.

**Review by ckittel — APPROVED** [review]
Thanks for the updates, LGTM

**Comment by ckittel** [comment]
@mmacy -- This is ready for your review.  Feel free to review the README and anything else you wish before initial merge.

File: src/protected-api-access-protected-api/README.md
Comment (mmacy) [inline src/protected-api-access-protected-api/README.md:184]:
```suggestion
This ASP.NET Core minimal web API has a single protected route, `/me`, that requires callers (client applications making requests to the endpoint) present a valid access token issued by the Microsoft identity platform.

Acting as a middle-tier API, this minimal web API then uses that access token to acquire a second access token from the Microsoft identity platform, this time for Microsoft Graph, on behalf of the user.

Finally, the web API requests data from the Microsoft Graph `/me` endpoint and includes the response data in its response to the original caller.
```

**Review by mmacy — APPROVED** [review]
LGTM. Bunch of README suggestions but nothing blocking. Thanks, @ckittel! :shipit: 

As an aside, it's crazy how lightweight the code is. This really goes to show how this is going to be more about explaining the two app registrations, scopes, perms, consent, and whathaveyou more than the actual code.

**Comment by ferantivero** [comment]
thanks a lot for your feedback/improvements over here @mmacy 💪 

 @ckittel, it sounds we've got green light to squash!

**Pull request merged**
